bignum_kilic_red.go

```diff
@@ -0,0 +1,116 @@
+// +build bignum_kilic_red
+
+package kate
+
+import (
+	"crypto/rand"
+	"encoding/binary"
+	"math/big"
+
+	kbls "github.com/kilic/bls12-381"
+)
+
+func init() {
+	initGlobals()
+	ClearG1(&ZERO_G1)
+	initG1G2()
+}
+
+type Big kbls.Fr
+
+func norm(in *Big) *Big {
+	e := new(Big)
```

Comment:
Why create a new big-number? Can we not just run `.FromRed()` on the input, and define it as `func norm(v *Big)`, changing it in-place? If the user of the function wants a copy, they can always copy it themselves.